src/formatDate.js

```diff
@@ -50,7 +50,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const oldDateSplit = date.split(oldSeparator);
+  const newDate = [];
+  const month = oldDateSplit[fromFormat.indexOf('MM')];
+  const day = oldDateSplit[fromFormat.indexOf('DD')];
+  let year;
+
+  if (fromFormat.includes('YYYY')) {
+    year = oldDateSplit[fromFormat.indexOf('YYYY')].slice(-2);
+  } else {
+    year = oldDateSplit[fromFormat.indexOf('YY')];
+  }
+
+  for (let newOrder = 0; newOrder < toFormat.length - 1; newOrder++) {
+    switch (toFormat[newOrder]) {
+      case 'YY':
+        newDate[newOrder] = year;
+        break;
+
+      case 'YYYY':
+        newDate[newOrder] = +year < 30 ? '20' + year : '19' + year;
+        break;
+
+      case 'MM':
+        newDate[newOrder] = month;
+        break;
+
+      case 'DD':
+        newDate[newOrder] = day;
+    }
```

Comment:
```suggestion
        newDate[newOrder] = day;
        break;
    }
```

Reply:
All `case` blocks must have a `break` statement. Also `switch/case` statement always must have a `default` block. In the `default` block, we usually raise exceptions if something went wrong. formatDate function can return a string "Get wrong date format" in the `default` block